From 6a98f0d1001cbeeaec02a59c946b0b8f9dc2127f Mon Sep 17 00:00:00 2001
From: Li Wencheng <liwencheng@phytium.com.cn>
Date: Fri, 26 Apr 2024 14:55:39 +0800
Subject: [PATCH 16/67] PHYTIUM: net/phytmac: Get device type error fixed

We should return -EOPNOTSUPP to the unsupported device types,
otherwise the wireless attribute will be added by default.

Mainline: NA
Signed-off-by: Li Wencheng <liwencheng@phytium.com.cn>
Signed-off-by: Wang Yinfeng <wangyinfeng@phytium.com.cn>
Change-Id: Ie0522ed5c744ddc885ab633dac5ecc9661c580f2

Link: https://gitee.com/phytium_embedded/phytium-linux-kernel/commit/66403a1703bf00c249e081103f7efbc2867c53bc
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/net/ethernet/phytium/phytmac_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/phytium/phytmac_main.c b/drivers/net/ethernet/phytium/phytmac_main.c
index ca1a03fa3552..058f9647b855 100644
--- a/drivers/net/ethernet/phytium/phytmac_main.c
+++ b/drivers/net/ethernet/phytium/phytmac_main.c
@@ -1843,7 +1843,7 @@ static int phytmac_close(struct net_device *ndev)
 static int phytmac_ioctl(struct net_device *dev, struct ifreq *rq, int cmd)
 {
 	struct phytmac *pdata = netdev_priv(dev);
-	int ret;
+	int ret = -EOPNOTSUPP;

 	if (!netif_running(dev))
 		return -EINVAL;
--
2.50.1
